app/src/main/java/com/example/android/codelabs/paging/ui/ReposAdapter.kt

```diff
@@ -21,37 +21,49 @@ import androidx.paging.PagingDataAdapter
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.RecyclerView.ViewHolder
 import com.example.android.codelabs.paging.R
-import com.example.android.codelabs.paging.model.Repo
 
 /**
  * Adapter for the list of repositories.
  */
-class ReposAdapter : PagingDataAdapter<Repo, ViewHolder>(REPO_COMPARATOR) {
+class ReposAdapter : PagingDataAdapter<UiModel, ViewHolder>(UIMODEL_COMPARATOR) {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-        return RepoViewHolder.create(parent)
+        return if (viewType == R.layout.repo_view_item) {
+            RepoViewHolder.create(parent)
+        } else {
+            SeparatorViewHolder.create(parent)
+        }
     }
 
     override fun getItemViewType(position: Int): Int {
-        return R.layout.repo_view_item
+        return when (getItem(position)) {
+            is UiModel.RepoItem -> R.layout.repo_view_item
+            is UiModel.SeparatorItem -> R.layout.separator_view_item
+            null -> 0
+        }
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val repo = getItem(position)
-        (holder as RepoViewHolder).bind(repo)
+        val uiModel = getItem(position)
+        uiModel.let {
+            when (uiModel) {
+                is UiModel.RepoItem -> (holder as RepoViewHolder).bind(uiModel.repo)
+                is UiModel.SeparatorItem -> (holder as SeparatorViewHolder).bind(uiModel.description)
+            }
+        }
     }
 
     companion object {
-        private val REPO_COMPARATOR = object : DiffUtil.ItemCallback<Repo>() {
-            override fun areItemsTheSame(oldItem: Repo, newItem: Repo): Boolean {
-                return if (oldItem is Repo && newItem is Repo) {
-                    oldItem == newItem
+        private val UIMODEL_COMPARATOR = object : DiffUtil.ItemCallback<UiModel>() {
+            override fun areItemsTheSame(oldItem: UiModel, newItem: UiModel): Boolean {
+                return if (oldItem is UiModel.RepoItem && newItem is UiModel.RepoItem) {
```

Comment:
What do you think about just returning 

`oldItem is UiModel.RepoItem && newItem is UiModel.RepoItem && oldItem.repo.fullName == newItem.repo.fullName` directly here?